File: Competitive Programming/codeforces/training/B_Diverse_Substrings.py
# Copyright (c) 2023, Le Duc Phuc Long

# If you don't think twice, you have to code twice.

# Import session
import sys
from collections import defaultdict
from math import *
input = sys.stdin.readline
output = sys.stdout.readline


############ ---- Input Functions ---- ############
def inp():
    return int(input())

def inlt():
    return list(map(int, input().split()))

def instr():
    return list(input().strip())

def invr():
    return map(int, input().split())


############ ---- Other Functions ---- ############
# Precompute

# Main function
# sys.stdin = open('in.txt', 'r')
for _ in range(inp()):
    n = inp()
    s = instr()
    
    ans = 0
    for i in range(n):
        # print("i: ", s[i])
        fr = [0]*(10)
        distinct, max_fre = 0, 0
        for j in range(i, min(i+100, n)):
            fr[int(s[j])] += 1
            if fr[int(s[j])] == 1:
                distinct += 1
            if fr[int(s[j])] == 11:
                break
            max_fre = max(max_fre, fr[int(s[j])])
            if max_fre <= distinct:
                ans += 1

    print(ans)
            

File: Competitive Programming/codeforces/training/B_Diverse_Substrings.cpp
// Copyright (c) 2023, Le Duc Phuc Long

/*If you don't think twice, you have to code twice.*/

// Include
#include <bits/stdc++.h>  // NOLINT
// #include <conio.h>

// Namespace
using namespace std;  // NOLINT

// Define
#define nln '\n'

// Typedef and Struct
typedef int64_t lo;

// Constant

// Declare global variables.
vector<pair<lo, lo>> dir4 = {{0, 1}, {0, -1}, {1, 0}, {-1, 0}};
// End of global variable declaration.

void precompute() {
}

void solve() {
    lo n;   cin >> n;
    string s;   cin >> s;
    lo ans = 0;
    for (lo i = 0; i < n; ++i) {
        lo max_fr = 0, distince = 0;
        map<lo, lo> fre;
        for (lo j = i; j < min(i+100, n); ++j) {
            lo x = s[j]-'0';
            fre[x]++;
            if (fre[x] == 11)
                break;
            if (fre[x] == 1)
                distince++;
            max_fr = max(max_fr, fre[x]);
            if (max_fr <= distince)
                ans++;
        }
    }
    cout << ans << nln;
}

int main(int argc, char* argv[]) {
    //freopen("in.txt", "r", stdin);
    //freopen("out.txt", "w", stdout);
    cin.tie(0)->sync_with_stdio(0);
    cout.tie(0)->sync_with_stdio(0);
    int T = 1;
    cin >> T;
    precompute();
    while (T--) {
        solve();
    }
    cerr << "It's ok Long, Good for now !!" << nln;
    return 0;
}